### DISCUSSION TITLE
having trouble adding tracks to Multitrack . please help

### DISCUSSION THREAD

**Discussion by djwugee** [katspaugh/wavesurfer.js#3582] [Q&A]
can someone please maybe help me with how the tracks are loaded into multitrack
here an example of a button to load a track into the timeline.

const addlayer = document.querySelector('#upload')
const uploadedAudioFile = document.getElementById('file-input');

addlayer.onclick = () => {
  uploadedAudioFile.click();
};

uploadedAudioFile.addEventListener('change', function({ id }) {
  
  const file = this.files[0];

  if (file) {
    const url = URL.createObjectURL(file);

    multitrack.addTrack({
      id,
      url: url,
      startPosition: 0,
      draggable: true,
      options: {
        waveColor: 'hsl(25, 87%, 49%)',
        progressColor: 'hsl(25, 87%, 20%)',
      },
    });
  }
});

however it clears the previous track and only loads in the top layer.

i just want to figure out how to load multiple uploaded  user tracks.
and have been have trouble..
any help would be appreciated

**Comment by katspaugh** [comment]
Make sure when you initialize multitrack that a track with that id already exists. `addTrack` can only add audio to a pre-existing empty track.

Please note that I don't offer free support for multitrack as stated in the [readme](https://github.com/katspaugh/wavesurfer-multitrack?tab=readme-ov-file#support). Replying to this one as an exception.
